src/nav/NavigationService.js

```diff
@@ -0,0 +1,55 @@
+/* @flow strict-local */
+import React from 'react';
+import type {
+  NavigationAction,
+  NavigationNavigatorProps,
+  NavigationState,
+  NavigationDispatch,
+  SupportedThemes,
+} from 'react-navigation';
+
+type ReduxContainerProps = {
+  ...$Diff<NavigationNavigatorProps<{}, NavigationState>, { navigation: mixed }>,
+  state: NavigationState,
+  dispatch: NavigationDispatch,
+  theme: SupportedThemes | 'no-preference',
+};
+
+// Should mimic return type of react-navigation-redux-helpers's
+// `createReduxContainer`
+//
+// TODO: This will eventually be a ref to the navigation component
+// instance, not the Redux container. See
+// https://reactnavigation.org/docs/4.x/navigating-without-navigation-prop.
+type ReduxContainer = React$Component<ReduxContainerProps>;
+
+const reduxContainerRef = React.createRef<ReduxContainer>();
+
+const getState = (): NavigationState => {
+  if (reduxContainerRef.current === null) {
+    throw new Error('Tried to use NavigationService before reduxContainerRef was set.');
+  }
+  return (
+    reduxContainerRef.current
+      // $FlowFixMe - how to tell Flow about this method?
+      .getCurrentNavigation().state
```

Comment:
> Looks like the existing libdef for `react-navigation-redux-helpers` is a stub, so that's no help. Given that we're aiming to rip that out soon, it may not be worth the effort to prepare a real one.

Makes sense. I expect we'll have a very similar problem with the ref we'll set to React Navigation's own component (when r-n-r-h's `createReduxContainer` out of the picture), but we can cross that bridge when we come to it.

Reply:
> Looks like the existing libdef for `react-navigation-redux-helpers` is a stub, so that's no help.

True, but there are some Flow types in node_modules/react-navigation-redux-helpers/src/create-redux-container.js. But I didn't glean anything that I found particularly helpful for this.